src/hotspot/share/opto/addnode.hpp

```diff
@@ -333,6 +399,26 @@ class MinINode : public MaxNode {
   virtual Node *Ideal(PhaseGVN *phase, bool can_reshape);
 };
 
+//------------------------------UMaxINode---------------------------------------
+// Maximum of 2 unsigned integers.
+class UMaxINode : public Node {
```

Comment:
Would it be better to define `max_opcode()` and `min_opcode()` for `UMaxINode` and `UMinINode`? These are used to find commutative patterns in `AddNode::Ideal()` and `MulNode::Ideal()` and optimize them.